numba/core/cpu.py

```diff
@@ -290,6 +292,11 @@ def finalize(self, flags, options):
         if not flags.is_set("debuginfo"):
             flags.debuginfo = config.DEBUGINFO_DEFAULT
 
+        if flags.debuginfo:
+            if not flags.is_set("dbg_extend_lifetimes"):
+                # auto turn on extend-lifetimes if debuginfo is on.
+                flags.dbg_extend_lifetimes = True
```

Comment:
Does this present a problem if the code being debugged is suspected to have a bug related to life times? Is it possible to instead do the above but also include: if the user explicitly sets  `dbg_extend_lifetimes` to `False` and `debug` to `True`, then life times are not extended. i.e. the explicit request to not extend lifetimes is acknowledged but the rest of `debuginfo` works as usual.

Reply:
Yes, this is what is happening. The comment needs updating.